src/commands/contract/test/index.ts

```diff
@@ -0,0 +1,194 @@
+import "ts-mocha";
+import { Flags, Args } from "@oclif/core";
+import path from "node:path";
+import { globby } from "globby";
+import Mocha from "mocha";
+import { emptyDir } from "fs-extra/esm";
+import shell from "shelljs";
+import { Contract } from "../../../lib/contract.js";
+import { SwankyCommand } from "../../../lib/swankyCommand.js";
+import { ConfigError, FileError, InputError, ProcessError, TestError } from "../../../lib/errors.js";
+import { spawn } from "node:child_process";
+import { Spinner } from "../../../lib/index.js";
+
+declare global {
+  var contractTypesPath: string; // eslint-disable-line no-var
+}
+
+export class TestContract extends SwankyCommand<typeof TestContract> {
+  static description = "Run tests for a given contact";
+
+  static flags = {
+    all: Flags.boolean({
+      default: false,
+      char: "a",
+      description: "Set all to true to compile all contracts",
+    }),
+    "local-node": Flags.boolean({
+      default: false,
+      description: "Run tests on a local node",
+    }),
+    mocha: Flags.boolean({
+        default: false,
+        description: "Run tests with mocha",
+    }),
+  };
+
+  static args = {
+    contractName: Args.string({
+      name: "contractName",
+      default: "",
+      description: "Name of the contract to test",
+    }),
+  };
+
+  async run(): Promise<void> {
+    const { args, flags } = await this.parse(TestContract);
+
+    if (args.contractName === undefined && !flags.all) {
+      throw new InputError("No contracts were selected to compile");
+    }
+
+    const contractNames = flags.all
+      ? Object.keys(this.swankyConfig.contracts)
+      : [args.contractName];
+
+    const testDir = path.resolve("tests");
```

Comment:
Move this to `--mocha` and test it throwing a FileError if no "tests" folder exists.